### PR TITLE
NMS-15341: cosign image and SBOM signing and attestation

### PR DESCRIPTION
**Pull request by RangerRick**
This PR implements docker image signing using [cosign](https://github.com/sigstore/cosign), an alternative to Docker Content Trust that supports signing other arbitrary resources including SBOMs (Software Bill-Of-Materials).

I have implemented SBOM-signing/attestation along with image signing.

Once approved, I will remove the temporary code that allowed me to publish from my branch to dockerhub and similar.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-15341

### CODE REVIEW

File: .circleci/main/workflows/workflows_v2.json [hunk 1]
```diff
@@ -509,6 +510,7 @@
             "filters": {
                 "branches": {
                     "only": [
+                        "jira/NMS-15341-cosign-sbom",
```
Comment (mershad-manesh) [inline .circleci/main/workflows/workflows_v2.json:513]: Do we need this filter?

**Comment by RangerRick** [comment]
Note that you can validate signing by using the public key (`cosign.pub`) uploaded to cloudsmith:

```
cosign verify --key https://packages.opennms.com/public/common/raw/files/cosign.pub docker.io/opennms/horizon:jira-NMS-15341-cosign-sbom-linux-amd64 | jq .

cosign verify-attestation --key https://packages.opennms.com/public/common/raw/files/cosign.pub docker.io/opennms/horizon:jira-NMS-15341-cosign-sbom-linux-amd64 | jq --slurp --compact-output -r 'map(.payload | @base64d | fromjson | .predicate.Data) | first' | less
```

**Review by mershad-manesh — APPROVED** [review]
To me the changes look good.

**Comment by RangerRick** [comment]
I'm gonna go ahead and merge this so we can try to get it into the 2023 release, but if any of y'all have feedback I'd love to see it still.

**Pull request merged**